ray-operator/test/sampleyaml/rayservice_test.go

```diff
@@ -0,0 +1,60 @@
+package sampleyaml
+
+import (
+	"testing"
+
+	. "github.com/onsi/gomega"
+
+	rayv1 "github.com/ray-project/kuberay/ray-operator/apis/ray/v1"
+	. "github.com/ray-project/kuberay/ray-operator/test/support"
+)
+
+func TestRayService(t *testing.T) {
+	tests := []struct {
+		name string
+	}{
+		{
+			name: "ray-service.sample.yaml",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			test := With(t)
+			g := NewWithT(t)
+
+			namespace := test.NewTestNamespace()
+			test.StreamKubeRayOperatorLogs()
+			rayServiceFromYaml := DeserializeRayServiceSampleYAML(test, tt.name)
+			KubectlApplyYAML(test, tt.name, namespace.Name)
+
+			rayService, err := GetRayService(test, namespace.Name, rayServiceFromYaml.Name)
+			g.Expect(err).NotTo(HaveOccurred())
+			g.Expect(rayService).NotTo(BeNil())
+
+			var rayClusterName string
+			// Wait for RayCluster name to be populated
+			g.Eventually(func() string {
+				rs, err := GetRayService(test, namespace.Name, rayServiceFromYaml.Name)
+				g.Expect(err).NotTo(HaveOccurred())
+				rayClusterName = rs.Status.PendingServiceStatus.RayClusterName
+				return rayClusterName
+			}, TestTimeoutShort).ShouldNot(BeEmpty())
```

Comment:
`RayClusterName` may not be always retrievable from `PendingServiceStatus`, you need to check both `PendingServiceStatus` and `ActiveServiceStatus`.

Although the probability is low, it's possible that it quickly changes from pending to active.